### PR TITLE
Making Agent Tests Arm compatible

### PR DESCRIPTION
**Pull request by shubham2892**
### Summary
Makes Agent Arm compatible. 

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/upload-images [hunk 1]
```diff
@@ -84,6 +84,13 @@ for image in "executionrole" "busybox" "nginx" "python" "ubuntu" "consul" "debia
   fi
 done
 
+REGISTRY_IMAGE_CONTAINER="test-ecs-registry-container:arm"
+REGISTRY_IMAGE="https://github.com/tianon/docker-registry-image.git#multiarch:2.6"
```
Comment (petderek) [inline scripts/upload-images]: is there an official repo for this?

Comment (shubham2892) [reply scripts/upload-images]: Yes they added support for Arm 😄  Updating.

File: agent/functional_tests/tests/functionaltests_unix_test.go [hunk 2]
```diff
@@ -536,6 +537,10 @@ func TestFluentdTag(t *testing.T) {
 	// tag was added in docker 1.9.0
 	RequireDockerVersion(t, ">=1.9.0")
 
+	if runtime.GOOS == "arm64" {
+		t.Skip()
```
Comment (sharanyad) [inline agent/functional_tests/tests/functionaltests_unix_test.go]: why are these tests skipped?

Comment (shubham2892) [reply agent/functional_tests/tests/functionaltests_unix_test.go]: Skipping this test for Arm as Fluentd docker container does not have official support for Arm.

**Pull request merged**